tests/test_ecu_variant_matching.py

```diff
@@ -0,0 +1,285 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2023 MBition GmbH
+
+import json
+from typing import Union
+
+import pytest
+
+from odxtools.diaglayer import DiagLayer
+from odxtools.diaglayertype import DIAG_LAYER_TYPE
+from odxtools.ecu_variant_matching import EcuVariantMatcher
+from odxtools.ecu_variant_patterns import EcuVariantPattern, MatchingParameter
+from odxtools.odxlink import OdxDocFragment, OdxLinkDatabase, OdxLinkId
+from odxtools.service import DiagService
+from odxtools.structures import Request, Response
+
+doc_frags = [OdxDocFragment("pytest", "WinneThePoh")]
+
+odxlinks = OdxLinkDatabase()
+
+
+@pytest.fixture
+def dummy_response(monkeypatch):
+    resp = Response(
+        odx_id=OdxLinkId(local_id="dummy_resp", doc_fragments=doc_frags),
+        short_name="dummy_resp",
+        parameters=None,
+    )
+    odxlinks.update({resp.odx_id: resp})
+
+    def decode(message: Union[bytes, bytearray]):
+        msg_str = message.decode(encoding="utf-8")
+        msg_dict = json.loads(msg_str)
+        return msg_dict
+
+    monkeypatch.setattr(resp, "decode", decode)
+    return resp
+
+
+@pytest.fixture()
+def identificationService(monkeypatch, dummy_response):
```

Comment:
python normally uses snake_case for function names:
```suggestion
def identification_service(monkeypatch, dummy_response):
```

(The python convention of using CamelCase for class names is a bit inconsistent IMO, but.... Also, I left out the other instances of camelCase as an exercise for the reader.)

Further, type annotations would be great here...